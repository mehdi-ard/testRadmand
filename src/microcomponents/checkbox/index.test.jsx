import React from 'react';
import renderer from 'react-test-renderer';
import Checkbox from '.';

it('renders defult', () => {
  expect(renderer.create(<Checkbox>test</Checkbox>).toJSON()).toMatchSnapshot();
});

it('renders default checked', () => {
  expect(
    renderer.create(<Checkbox checked>test</Checkbox>).toJSON()
  ).toMatchSnapshot();
});

it('renders disabled', () => {
  expect(
    renderer.create(<Checkbox disabled={true}>disabled</Checkbox>).toJSON()
  ).toMatchSnapshot();
});
it('renders indeterminate', () => {
  expect(
    renderer
      .create(<Checkbox indeterminate={true}>indeterminate</Checkbox>)
      .toJSON()
  ).toMatchSnapshot();
});
