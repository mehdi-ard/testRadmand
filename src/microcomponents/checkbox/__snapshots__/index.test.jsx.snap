// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders default checked 1`] = `
<label
  className="ant-checkbox-wrapper ant-checkbox-wrapper-checked"
>
  <span
    className="ant-checkbox ant-checkbox-checked"
    style={Object {}}
  >
    <input
      checked={true}
      className="ant-checkbox-input"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      type="checkbox"
    />
    <span
      className="ant-checkbox-inner"
    />
  </span>
  <span>
    test
  </span>
</label>
`;

exports[`renders defult 1`] = `
<label
  className="ant-checkbox-wrapper"
>
  <span
    className="ant-checkbox"
    style={Object {}}
  >
    <input
      checked={false}
      className="ant-checkbox-input"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      type="checkbox"
    />
    <span
      className="ant-checkbox-inner"
    />
  </span>
  <span>
    test
  </span>
</label>
`;

exports[`renders disabled 1`] = `
<label
  className="ant-checkbox-wrapper ant-checkbox-wrapper-disabled"
>
  <span
    className="ant-checkbox ant-checkbox-disabled"
    style={Object {}}
  >
    <input
      checked={false}
      className="ant-checkbox-input"
      disabled={true}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      type="checkbox"
    />
    <span
      className="ant-checkbox-inner"
    />
  </span>
  <span>
    disabled
  </span>
</label>
`;

exports[`renders indeterminate 1`] = `
<label
  className="ant-checkbox-wrapper"
>
  <span
    className="ant-checkbox ant-checkbox-indeterminate"
    style={Object {}}
  >
    <input
      checked={false}
      className="ant-checkbox-input"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      type="checkbox"
    />
    <span
      className="ant-checkbox-inner"
    />
  </span>
  <span>
    indeterminate
  </span>
</label>
`;
